src/apis/Style.res

```diff
@@ -154,20 +154,146 @@ type justifyContent = [
+  textDecorationColor?: Color.t,
+  textDecorationLine?: textDecorationLine,
+  textDecorationStyle?: textDecorationStyle,
+  textShadowColor?: Color.t,
+  textShadowOffset?: offset,
+  textShadowRadius?: float,
+  textTransform?: textTransform,
+  writingDirection?: writingDirection,
+  // View styles https://reactnative.dev/docs/view-style-props
+  backfaceVisibility?: backfaceVisibility,
+  backgroundColor?: Color.t,
+  borderBottomColor?: Color.t,
+  borderBottomEndRadius?: float,
+  borderBottomLeftRadius?: float,
+  borderBottomRightRadius?: float,
+  borderBottomStartRadius?: float,
+  borderBottomWidth?: float,
+  borderColor?: Color.t,
+  borderEndColor?: Color.t,
+  borderEndWidth?: float,
+  borderLeftColor?: Color.t,
+  borderLeftWidth?: float,
+  borderRadius?: float,
+  borderRightColor?: Color.t,
+  borderRightWidth?: float,
+  borderStartColor?: Color.t,
+  borderStartWidth?: float,
+  borderStyle?: borderStyle,
+  borderTopColor?: Color.t,
+  borderTopEndRadius?: float,
+  borderTopLeftRadius?: float,
+  borderTopRightRadius?: float,
+  borderTopStartRadius?: float,
+  borderTopWidth?: float,
+  borderWidth?: float,
+  elevation?: float,
+  opacity?: float,
+  // Transform Props (https://reactnative.dev/docs/transforms#props)
+  transform?: array<transform>, // all other transform props are deprecated
+  // Shadow Props (https://reactnative.dev/docs/shadow-props)
+  shadowColor?: Color.t,
+  shadowOffset?: offset,
+  shadowOpacity?: float,
+  shadowRadius?: float,
+  // Layout Style Props (https://reactnative.dev/docs/layout-props)
+  alignContent?: alignContent,
+  alignItems?: alignItems,
+  alignSelf?: alignSelf,
+  aspectRatio?: float,
+  // border*Width are commented because already in view styles props (see explanation at the top)
+  // borderBottomWidth?: float,
+  // borderEndWidth?: float,
+  // borderLeftWidth?: float,
+  // borderRightWidth?: float,
+  // borderStartWidth?: float,
+  // borderTopWidth?: float,
+  // borderWidth?: float,
+  bottom?: size,
+  direction?: direction,
+  display?: display,
+  end?: size,
+  flex?: float,
+  flexBasis?: margin,
+  flexDirection?: flexDirection,
+  flexGrow?: float,
+  flexShrink?: float,
+  flexWrap?: flexWrap,
+  height?: size,
+  justifyContent?: justifyContent,
+  left?: size,
+  margin?: margin,
+  marginBottom?: margin,
+  marginEnd?: margin,
+  marginHorizontal?: margin,
+  marginLeft?: margin,
+  marginRight?: margin,
+  marginStart?: margin,
+  marginTop?: margin,
+  marginVertical?: margin,
+  maxHeight?: size,
+  maxWidth?: size,
+  minHeight?: size,
+  minWidth?: size,
+  overflow?: overflow,
+  padding?: size,
+  paddingBottom?: size,
+  paddingEnd?: size,
+  paddingHorizontal?: size,
+  paddingLeft?: size,
+  paddingRight?: size,
+  paddingStart?: size,
+  paddingTop?: size,
+  paddingVertical?: size,
+  position?: position,
+  right?: size,
+  start?: size,
+  top?: size,
+  width?: size,
+  zIndex?: int,
+}
+
+external styleObj: styleObj => t = "%identity"
```

Comment:
Could we have `styleObj` type being directly `t` so `styleObj` external is not necessary ? Most people probably use `style()` so we could just use the record directly.
Also while we are changing things for style, what do you think about https://github.com/rescript-react-native/rescript-react-native/discussions/750 ?

Reply:
I thought about this, too, but the problem is that `t` may also be an array, so runtime errors could then be produced with things like

```rescript
Style.array([]).color
```

I think it is correct to keep `t` as an abstract type.

Reply:
As for #750, personally I prefer the way it is currently implemented.